test/regression/aarch64/Syscall.cc

```diff
@@ -187,7 +187,15 @@ TEST_P(Syscall, faccessat) {
   unlink(filepath);
 
   char abs_filepath[LINUX_PATH_MAX];
-  realpath(SIMENG_AARCH64_TEST_ROOT "/data/input.txt", abs_filepath);
+  char* output =
+      realpath(SIMENG_AARCH64_TEST_ROOT "/data/input.txt", abs_filepath);
```

Comment:
Prevent scope-leak: 
```
if (char* output = realpath(SIMENG_AARCH64_TEST_ROOT "/data/input.txt", abs_filepath); !output){
  std::cerr << "[SimEng:syscall] realpath failed with errno = " << errno
              << std::endl;
  exit(EXIT_FAILURE);
}
```

Reply:
This solution looks a bit horrible to me but I see your point. @jj16791 @FinnWilkinson @JosephMoore25 @ABenC377

Reply:
What's the base issue here?

Reply:
The base issue is that you get warnings if you don't check the output of realpath

Reply:
Alex's suggestion seems to work so updated in all places